src/VecSim/spaces/L2/L2_SVE2_UINT8.h

```diff
@@ -0,0 +1,105 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "VecSim/spaces/space_includes.h"
+#include <arm_sve.h>
+
+// Aligned step using svptrue_b8()
+inline void L2SquareStep(const uint8_t *&pVect1, const uint8_t *&pVect2, size_t &offset,
+                         svint32_t &sum, const size_t chunk) {
+    svbool_t pg = svptrue_b8();
+    // Note: Because all the bits are 1, the extention to 16 and 32 bits does not make a difference
+    // Otherwise, pg should be recalculated for 16 and 32 operations
+
+    svuint8_t v1_i8 = svld1_u8(pg, pVect1 + offset); // Load int8 vectors from pVect1
+    svuint8_t v2_i8 = svld1_u8(pg, pVect2 + offset); // Load int8 vectors from pVect2
+
+    // Subtract v2 from v1 and widen the results to int16 for the even indexes
+    svint16_t diff_e = svreinterpret_s16(svsublb_u16(v1_i8, v2_i8));
```

Comment:
is reinterpret cast the correct type of cast here? (and in all other L2 implementations)

Reply:
As we discussed, it's ok to reinterpret,  but adding comment to explain and  unit_test for future cases